### DISCUSSION TITLE
How render controls of parent group?

### DISCUSSION THREAD

**Discussion by vshepel** [fabricjs/fabric.js#9780] [Q&A] [answered]
I want to draw group borders when a sub target selected (so that the user can see the boundaries of group).

This code worked, but I think this is the wrong way to implement this task:

```ts
canvas.on('selection:created', () => {
    const group = canvas.getActiveObject().parent

    // Without timeout not drawed 
    setTimeout(() => {
      group._renderControls(editor.getContext(), { hasControls: false })
    })
  })
  ```
  
Result: 
  
<img width="206" alt="image" src="https://github.com/fabricjs/fabric.js/assets/14239290/13e71255-3bbd-41a3-8d24-f4064084151b">

**Comment by asturur** [comment, marked as answer]
setTimeout isn't great, no.
Did you try use the 'after:render' event?
The Issue would be that since your controls have been already rendered the group ones will be renderered on top of it.
That may be a deal breaker or not i m not sure.

Let me see if there is some kind of other event that can be used

**Reply by asturur** [reply]
What version of fabric are you using?

**Reply by vshepel** [reply]
With 'after:render' it looks great!
I tried different events earlier, but missed the given one )

Many thanks!
